### DISCUSSION TITLE
Comparing with obsession

### DISCUSSION THREAD

**Discussion by timsofteng** [rmagatti/auto-session#128] [General]
Hello.
May you please briefly compair auto-session and obsession from tpope?
I use second one. Should I switch?

**Comment by rmagatti** [comment]
Well @timsofteng switching depends on what you want out of a session plugin. tpope automated sessions for his workflow, I automated them for mine.

My workflow consists in sessions automatically loading up where session names are automatically created based on the directory one opened vim in. i.e `cd Projects/my-project && nvim` tries to load a session called `my-project` (it's actually a full path for uniqueness), if it doesn't find one it will still create a session for that unique path upon vim exit, which is 99.9% of the time what I want it to do. From what I understand from tpope's session plugin, it
1. relies on the user manually choosing to create and load a session 
2. creates the session file on the project dir (auto-session stores sessions centrally)

So Auto Session essentially has a more automated approach compared to vim-obsession, it of course assumes a particular workflow (mine) and works best for how I'd like sessions to work .

To summarize, if you don't mind controlling the creation and loading of sessions yourself vim-obsession is good for you and you should stick with it. Now if you want to just not have to think about sessions but have things work just like they would in something like vscode or other editors, auto-session is likely the more appealing choice.

**Comment by timsofteng** [comment]
@rmagatti Thanks
The main advantage for me is ability to store all session in specific dir. Obsession can do it manually only.
Is it possible to swap logic in `auto-session` to change `nvim` and `nvim .`behavior? And start session inside neovim manually only.

**Reply by rmagatti** [reply]
Yeah you can disable specific behaviour.
```
auto_save_enabled = nil, -- Enables/disables auto save feature
auto_restore_enabled = nil, -- Enables/disables auto restore feature
```
